### DISCUSSION TITLE
Grouped chart with offset, but center if only one group

### DISCUSSION THREAD

**Discussion by jonchar** [vega/vega-lite#9452] [Q&A] [answered]
Hi - I'm looking to make a [grouped bar chart](https://vega.github.io/vega-lite/examples/bar_grouped.html) however when there are fewer than the max number of groups, I'd like to center those bars.

For example (assuming the data in the grouped bar chart example), if category B had only data from group x, then I would like the blue bar to be centered at the x-axis marker for category B (where the bar for y is in the example). I'd also like the bar width to be the same.

Any advice on how to accomplish this? Thank you!

**Comment by kgoodrick-uu** [comment, marked as answer]
Hi Jon,

I think there are two ways to achieve this, one isn't quite a grouped bar chart but doesn't need manual bar sizes and the other is a grouped bar chart but requires manual sizing of the bar. 

The first approach is to use a column encoding of an x encoding like [this](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoKGvJsATgE80IAIKKQZC2wY5bAD0fbdcNAAZKAEYAXwUTMzJLG3QHJWdXd3QbJR89AMoANjCIzHNrWzinFzdbKxSQNL9UQOzwkFM8qILYxwTS9AAvTu8ddJqsnIbI6MK2kqSQHtS+6sCATiHG-JiQACFxxM9e3wyAdiWRlvXNjpAKqoyAJkOm0fQAYVPJrxndgbrcldsAEWftt79QKherLZqrP7xCZlHZAwago4Q-7JC6zDKfYZ3Y6Q4pbLqwuaURYAXTCIGQFgA1mhQJgrDhqiAmEgLCAyXBZFA2MoaLIyDSGmxdAh5JoQAAzGhwQTKWxg0ZKeBIZRwVlizB0QRwADyFilskwtiYbGwbAQjkESCY0t1+sN6GNpvNITJMVAkulsvQlyUdIZtgAjgwkAa6GoaKQ2UovGKPTLbO13EokB4aAZULIGIJBGSuYJLAK417caUXUoLPohZGxdAdNVQDGQLyVQzZCqDWyXUA), which produces this chart:
<img width="419" alt="image" src="https://github.com/user-attachments/assets/63d72b89-ff78-4b4c-9148-09a21b42290e">

For the second we can use a series of transforms to generate manual xOffsets like [this](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoKGvJsATgE80IAIKKQZC2wY5bAD0fbdcNAAZKAEYAXwUTMzJLG3QHJWdXd3QbJR89AMoANjCIzHNrWzinFzdbKxSQNL9UQOzwkFM8qILYxwTS9AAvTu8ddJqsnIbI6MK2kqSQHtS+6sCATiHG-JiQACFxxM9e3wyAdiWRlvXNjpAKqoyAJkOm0fQAYVPJrxndgbrcldsAEWftt79QKherLZqrP7xCZlHZAwago4Q-7JC6zDKfYZ3Y6Q4pbLqwuaURYAXTCIGQFgA1mhQJgrDhqiAmEgLI4dDQyPJ0FA4LI8qylAB3GjKehoIL+MmYCxIWQQABmlgQRlAwtkyjYgpVIDYkxlsmpSnlNDggmUtna7iUSAM6H11JCxKhiSYMUMmO+pPqkjYNFkSDIzj4eW1utsCDgsscxtN5rtssNIBttntAH0I1HHc63K6jB7wSAvRFBFAGIIzLYABSqTAMBCUe0AAn4jZrdYbCfTkdkAEo2baQB4APLy+UQOCYEBZkC8qBsZR+sg0wfLmNm2xg0ZkmKgNdxypopR0hm2ACODFlmDoahopCnSmHo-Hk80ID3nhHY4njmPjPPl+vK87zJOdBEsVcTXXdBLSnMlhVFGA0CuMRJSAA), which produces this chart:
<img width="357" alt="image" src="https://github.com/user-attachments/assets/5f5b004d-aac3-4ea3-a016-77673e5da5ca">

-Kyle

**Reply by jonchar** [reply]
Thank you, this is exactly what I was looking for!
